stablehlo/reference/Ops.cpp

```diff
@@ -81,35 +81,20 @@ Tensor evalFloorOp(const Tensor &operand, Type resultType) {
 
 Tensor evalIotaOp(int64_t iotaDimension, Type resultType) {
   Tensor result(resultType);
-  Type elType = result.getType().getElementType();
+  Type elementTy = result.getType().getElementType();
   for (auto it = result.index_begin(); it != result.index_end(); ++it) {
-    auto iota = (*it)[iotaDimension];
-    if (isSupportedSignedIntegerType(elType)) {
-      result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
-                                            iota, /*isSigned=*/true)));
-    } else if (isSupportedUnsignedIntegerType(elType)) {
-      result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
-                                            iota, /*isSigned=*/false)));
-    } else if (isSupportedFloatType(elType)) {
-      APFloat val = APFloat((double)iota);
-      bool roundingErr;
-      val.convert(elType.cast<FloatType>().getFloatSemantics(),
-                  APFloat::rmNearestTiesToEven, &roundingErr);
-      result.set(*it, Element(elType, val));
-    } else if (isSupportedComplexType(elType)) {
-      APFloat real((double)iota);
-      APFloat imag((double)0.0);
-      FloatType flType =
-          elType.cast<ComplexType>().getElementType().cast<FloatType>();
-      bool roundingErr;
-      real.convert(flType.getFloatSemantics(), APFloat::rmNearestTiesToEven,
-                   &roundingErr);
-      imag.convert(flType.getFloatSemantics(), APFloat::rmNearestTiesToEven,
-                   &roundingErr);
-      result.set(*it, Element(elType, std::complex<APFloat>(real, imag)));
+    if (isSupportedIntegerType(elementTy)) {
+      result.set(*it, Element::getValue(elementTy, (*it)[iotaDimension]));
+    } else if (isSupportedFloatType(elementTy)) {
+      result.set(*it, Element::getValue(elementTy, static_cast<double>(
+                                                       (*it)[iotaDimension])));
+    } else if (isSupportedComplexType(elementTy)) {
+      result.set(*it, Element::getValue(
+                          elementTy, std::complex<double>(static_cast<double>(
```

Comment:
Use `std::complex<double>(static_cast<double>(...), 0.0)` instead? Juts to be transparent to the choice of imaginary value and explicitly maintaining parity with spec.